ir/src/integrity_stmts/graph.rs

```diff
@@ -171,7 +171,12 @@ impl AlgebraicGraph {
                 let (lhs, trace_segment, domain) =
                     self.insert_expr(symbol_table, *lhs, variable_roots)?;
                 // add exponent subexpression.
-                let node_index = self.insert_op(Operation::Exp(lhs, rhs as usize));
+                let node_index = if let Expression::Const(rhs) = *rhs {
+                    self.insert_op(Operation::Exp(lhs, rhs as usize))
+                } else {
+                    todo!()
+                };
+
```

Comment:
what's the reasoning for not completing this now? do we need to do something other than call `insert_expr` again for the `rhs`? I'd prefer to implement it now unless there's something complicated that I'm overlooking

Reply:
@grjte I haven't changed the signature of the `Exp` variant of the `Operation` enum yet because we will have to either evaluate exponent expressions or limit them to be able to calculate degree of the operation correctly. I was thinking to handle it with adding list comprehension to IR PR.

Reply:
Ok sure, makes sense.

Reply:
Actually, it can you add a TODO in the `Operation` enum as well before merging this?